File: php/example_code/ec2/DescribeAvailabilityZones.php
<?php
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

/*
 *  ABOUT THIS PHP SAMPLE: This sample is part of the SDK for PHP Developer Guide topic at
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/ec2-examples-using-security-groups.html
 *
 *
 *
 */
// snippet-start:[ec2.php.describe_availability_zones.complete]
// snippet-start:[ec2.php.describe_availability_zones.import]

require 'vendor/autoload.php';

// snippet-end:[ec2.php.describe_availability_zones.import]
/**
 * Describe Availability Zones
 *
 * This code expects that you have AWS credentials set up per:
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials.html
 */

// snippet-start:[ec2.php.describe_availability_zones.main]
$ec2Client = new Aws\Ec2\Ec2Client([
    'region' => 'us-west-2',
    'version' => '2016-11-15',
    'profile' => 'default'
]);

$result = $ec2Client->describeAvailabilityZones();

var_dump($result);

// snippet-end:[ec2.php.describe_availability_zones.main]
// snippet-end:[ec2.php.describe_availability_zones.complete]

File: cpp/example_code/ec2/describe_availability_zones.cpp
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
/**
 * Before running this C++ code example, set up your development environment, including your credentials.
 *
 * For more information, see the following documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started.html
 *
 * For information on the structure of the code examples and how to build and run the examples, see
 * https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started-code-examples.html.
 *
 **/

#include <aws/ec2/EC2Client.h>
// snippet-start:[ec2.cpp.describe_zones.inc]
#include <aws/ec2/model/DescribeAvailabilityZonesRequest.h>
// snippet-end:[ec2.cpp.describe_zones.inc]
#include <iomanip>
#include <iostream>
#include "ec2_samples.h"

// snippet-start:[cpp.example_code.ec2.DescribeAvailabilityZones]

//! DescribeAvailabilityZones
/*!
  \param clientConfiguration: AWS client configuration.
  \return bool: Function succeeded.
*/
int AwsDoc::EC2::describeAvailabilityZones(const Aws::Client::ClientConfiguration &clientConfiguration) {
    Aws::EC2::EC2Client ec2Client(clientConfiguration);
    // snippet-start:[ec2.cpp.describe_zones.code]
    Aws::EC2::Model::DescribeAvailabilityZonesRequest request;
    Aws::EC2::Model::DescribeAvailabilityZonesOutcome outcome = ec2Client.DescribeAvailabilityZones(request);

    if (outcome.IsSuccess()) {
        std::cout << std::left <<
                  std::setw(32) << "ZoneName" <<
                  std::setw(20) << "State" <<
                  std::setw(32) << "Region" << std::endl;

        const auto &zones =
                outcome.GetResult().GetAvailabilityZones();

        for (const auto &zone: zones) {
            Aws::String stateString =
                    Aws::EC2::Model::AvailabilityZoneStateMapper::GetNameForAvailabilityZoneState(
                            zone.GetState());
            std::cout << std::left <<
                      std::setw(32) << zone.GetZoneName() <<
                      std::setw(20) << stateString <<
                      std::setw(32) << zone.GetRegionName() << std::endl;
        }
    } else {
        std::cerr << "Failed to describe availability zones:" <<
                  outcome.GetError().GetMessage() << std::endl;

    }
    // snippet-end:[ec2.cpp.describe_zones.code]

    return outcome.IsSuccess();
}
// snippet-end:[cpp.example_code.ec2.DescribeAvailabilityZones]

/*
 *  main function
 *
 *  Usage:
 *        describe_availability_zones
 */

#ifndef TESTING_BUILD

int main() {
    Aws::SDKOptions options;
    InitAPI(options);

    {
        Aws::Client::ClientConfiguration clientConfig;
        // Optional: Set to the AWS Region (overrides config file).
        // clientConfig.region = "us-east-1";
        AwsDoc::EC2::describeAvailabilityZones(clientConfig);
    }

    ShutdownAPI(options);
    return 0;
}

#endif // TESTING_BUILD


